### PR TITLE
[O2B-1062] GET API LOG should allow for exclusion filter

### PR DESCRIPTION
**Pull request by DKR2002**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [ ] PR labels are selected

Notable changes for users:
- The exclusion filter was only applied to the author filter.
- The exclusion filter sign is set to an exclamation mark (!). Call the exclusion filter for the author, by placing this exclusion filter sign in front of the author you want to exclude (with or without a space).

Notable changes for developers:
- Added API test to test the new feature.
- For now the exclusion sign is set to an exclamation mark (!). This could be changed in the file itself (lib/usecases/log/GetAllLogsUseCase.js)

Changes made to the database:
-

### CODE REVIEW

File: lib/usecases/log/GetAllLogsUseCase.js [hunk 1]
```diff
@@ -29,6 +29,27 @@ const { ApiConfig } = require('../../config/index.js');
...
+ * removes the exclusion sign, if present.
+ *
+ * @param {string} dtoFilter the value to check for an exclusion sign.
+ * @returns {object} represents the parsed result.
+ * @property {boolean} excl indicates whether the first character of dtoFilter
+ *      is equal to exclSign.
+ * @property {string} strippedFilter the filtered string with the first
+ *      character removed if excl is true, otherwise, the original dtoFilter.
+ */
+const checkForFilterExclusion = (dtoFilter) => {
```
Comment (graduta) [inline lib/usecases/log/GetAllLogsUseCase.js:43]: I would recommend that you add this method as a general method in a separate file so that other services can potentially re-use it

**Review by graduta — CHANGES_REQUESTED** [review]

**Review by graduta — CHANGES_REQUESTED** [review]
Hi @DKR2002 ,

PR looks almost complete but I believe you should tackle the corner case in which the user types `!` but nothing else. 
In this scenario, you should still return all logs which is not the case with your changes now. Moreover, make sure you add a test for it

**Review by graduta — APPROVED** [review]
